homework/hw2-goals-approaches/ez/hw2-ez.md

```diff
@@ -0,0 +1,390 @@
+
+```{code-cell} ipython3
+!dvc pull
+```
+
+Load the data using pandas+pyarrow:
+
+```{code-cell} ipython3
+df = pd.read_feather('../../../data/mtg.feather')
+df.head()[['name','text', 'mana_cost', 'flavor_text','release_date', 'edhrec_rank']]
+```
+
+### Submission Structure
+You will need to submit a pull-request on DagsHub with the following additions: 
+
+- your subfolder, e.g. named with your user id, inside the `homework/hw2-goals-approaches/` folder
+    - your "lab notebook", as an **`.ipynb` or `.md`** (e.g. jupytext), that will be exported to PDF for Canvas submission. **This communicates your _goals_**, along with the results that will be compared to them. 
+    - your **`dvc.yaml`** file that will define  the inputs and outputs of your _approaches_. See [the DVC documentation](https://dvc.org/doc/user-guide/project-structure/pipelines-files) for information!
+    - **source code** and **scripts** that define the preprocessing and prediction `Pipeline`'s you wish to create. You may then _print_ the content of those scripts at the end of your notebook e.g. as appendices using 
+- any updates to `environment.yml` to add the dependencies you want to use for this homework
+
++++
+
+## Part 1: Unsupervised Exploration
+
+Investigate the [BERTopic](https://maartengr.github.io/BERTopic/index.html) documentation (linked), and train a model using their library to create a topic model of the `flavor_text` data in the dataset above. 
+
+- In a `topic_model.py`, load the data and train a bertopic model. You will `save` the model in that script as a new trained model object
+- add a "topic-model" stage to your `dvc.yaml` that has `mtg.feather` and `topic_model.py` as dependencies, and your trained model as an output
+- load the trained bertopic model into your notebook and display
+    1. the `topic_visualization` interactive plot [see docs](https://maartengr.github.io/BERTopic/api/plotting/topics.html)
+    2. Use the plot to come up with working "names" for each major topic, adjusting the _number_ of topics as necessary to make things more useful. 
+    3. Once you have names, create a _Dynamic Topic Model_ by following [their documentation](https://maartengr.github.io/BERTopic/getting_started/topicsovertime/topicsovertime.html). Use the `release_date` column as timestamps. 
+    4. Describe what you see, and any possible issues with the topic models BERTopic has created. **This is the hardest part... interpreting!**
+
+```{code-cell} ipython3
+from bertopic import BERTopic
+```
+
+```{code-cell} ipython3
+# Load the BERTopic model
+ft_model = BERTopic.load("ft_model")
+```
+
+```{code-cell} ipython3
+# Visualize topics
+ft_model.visualize_topics(top_n_topics = 9)
+```
+
+![dt_model.png](attachment:dt_model.png)
+
+```{code-cell} ipython3
+# Get topic information
+ft_model.get_topic_info()[2:10].set_index("Topic")
+```
+
+#### Renaming Topics
+Topic 1: kami <br>
+Topic 2: goblin <br>
+Topic 3: wildspeaker <br>
```

Comment:
Great insight. Would love to see a brief bullet list of why each of these makes sense (like a shorter version of your great Kami discussion, e.g. one bullet per topic, very short)